src/cart/crt_rpc.c

```diff
@@ -1344,14 +1349,22 @@ crt_req_send_internal(struct crt_rpc_priv *rpc_priv)
 			/* send addr lookup req */
 			rpc_priv->crp_state = RPC_STATE_ADDR_LOOKUP;
 			rc = crt_req_hg_addr_lookup(rpc_priv);
-			if (rc != 0)
+			if (rc == 0)
+				D_GOTO(send_immediately, rc);
+			else
 				D_ERROR("crt_req_hg_addr_lookup() failed, "
 					"rc %d, opc: %#x.\n", rc, req->cr_opc);
 		}
 		break;
 	case RPC_STATE_ADDR_LOOKUP:
+send_immediately:
 		rc = crt_req_send_immediately(rpc_priv);
```

Comment:
not sure i like the label in the middle of the case statement. can we just create the outside label and jump there instead?